worker/src/schedule.rs

```diff
@@ -0,0 +1,69 @@
+use std::future::Future;
+use std::process::Output;
+use worker_sys::{ScheduledEvent as EdgeScheduledEvent, ScheduleContext as EdgeScheduleContext};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::future_to_promise;
+
+/// [Schedule](https://developers.cloudflare.com/workers/runtime-apis/scheduled-event#syntax-module-worker)
+#[derive(Debug, Clone)]
+pub struct ScheduledEvent {
+    cron: String,
+    scheduled_time: f64,
+    ty: String,
+}
+
+impl From<EdgeScheduledEvent> for ScheduledEvent {
+    fn from(schedule: EdgeScheduledEvent) -> Self {
+        Self {
+            cron: schedule.cron(),
+            scheduled_time: schedule.scheduled_time(),
+            ty: String::from("scheduled"),
+        }
+    }
+}
+
+impl ScheduledEvent {
+    /// get cron trigger
+    pub fn cron(&self) -> String {
+        self.cron.clone()
+    }
+
+    /// get type
+    pub fn ty(&self) -> String {
+        self.ty.clone()
+    }
+
+    /// get trigger time as f64
+    pub fn schedule(&self) -> f64 {
+        self.scheduled_time.clone()
+    }
+}
+
+#[derive(Clone)]
+pub struct ScheduleContext {
+    edge: EdgeScheduleContext,
+}
+
+impl From<EdgeScheduleContext> for ScheduleContext {
+    fn from(edge: EdgeScheduleContext) -> Self {
+        Self {
+            edge
+        }
+    }
+}
+
+impl ScheduleContext {
+    pub fn wait_until<T>(&self, handler: T)
+        where
+            T: Future<Output=Result<(), ()>> + 'static,
+    {
+        self.edge.wait_until(
+            future_to_promise(
+                async {
+                    handler.await;
```

Comment:
This gives a warning because the result is ignored, we should either not use a result or throw it up to JavaScript.

Reply:
Oh yea, seems like I missed the result. Regarding [the PR](https://github.com/cloudflare/workers-rs/pull/109/files#diff-e2d4e1abd1defdb178fd60c050a700788731a1e9971e3188cc996e573b28c649R29) I removed it completely.